File: inspect_excel.mjs
import * as xlsx from 'xlsx';

try {
  const workbook = xlsx.readFile('enterprise-attack-v19.1.xlsx');
  console.log("Sheets:", workbook.SheetNames);
  
  if (workbook.SheetNames.includes('relationships')) {
    const relSheet = workbook.Sheets['relationships'];
    const relData = xlsx.utils.sheet_to_json(relSheet);
    if (relData.length > 0) {
      console.log("\nRelationships columns:", Object.keys(relData[0]));
      console.log("Sample relationships:", relData.slice(0, 3));
    }
  }

  if (workbook.SheetNames.includes('software')) {
    const softSheet = workbook.Sheets['software'];
    const softData = xlsx.utils.sheet_to_json(softSheet);
    if (softData.length > 0) {
      console.log("\nSoftware columns:", Object.keys(softData[0]));
      console.log("Sample software:", softData.slice(0, 3));
    }
  }
} catch (e) {
  console.error("Error:", e);
}

File: inspect_excel.py
import pandas as pd
import sys

file_path = "enterprise-attack-v19.1.xlsx"

try:
    xl = pd.ExcelFile(file_path)
    print("Sheets:", xl.sheet_names)
    
    if 'relationships' in xl.sheet_names:
        df_rel = xl.parse('relationships')
        print("\nRelationships columns:", df_rel.columns.tolist())
        print("Sample relationships:")
        print(df_rel.head(3).to_dict('records'))

    if 'software' in xl.sheet_names:
        df_soft = xl.parse('software')
        print("\nSoftware columns:", df_soft.columns.tolist())
        print("Sample software:")
        print(df_soft.head(3).to_dict('records'))
        
except Exception as e:
    print("Error:", e)
